src/hypertable.c

```diff
@@ -1187,6 +1188,31 @@ ts_hypertable_insert_blocker_trigger_add(PG_FUNCTION_ARGS)
 	PG_RETURN_OID(insert_blocker_trigger_add(relid));
 }
 
+static Datum
+create_hypertable_datum(FunctionCallInfo fcinfo, Hypertable *ht)
+{
+	TupleDesc	tupdesc;
+	Datum		values[3];
+	bool		nulls[3] = {false};
+	HeapTuple	tuple;
+
+	if (get_call_result_type(fcinfo, NULL, &tupdesc) != TYPEFUNC_COMPOSITE)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("function returning record called in "
+						"context that cannot accept type record")));
+
+	tupdesc = BlessTupleDesc(tupdesc);
+	values[0] = Int32GetDatum(ht->fd.id);
```

Comment:
Also here we could make these indexes more descriptive. E.g., creating an enum like in `catalog.h`